git_machete/git_operations.py

```diff
@@ -206,8 +208,9 @@ def __init__(self) -> None:
 
     @staticmethod
     def _run_git(git_cmd: str, *args: str, **kwargs: Any) -> int:
+        allow_non_zero = kwargs.pop("allow_non_zero", False)
```

Comment:
Different Bug, we have to pop this kwarg or the below `tils.run_cmd("git", git_cmd, *args, **kwargs)` would throw an error, same as `def _popen_git(git_cmd: str, *args: str, **kwargs: Any) -> str:`

Reply:
Hmmm for clarity... could you show the stack trace for this error? 🤔 is it coming from `subprocess.call`?

Reply:
While you're at this, pls also add some test that would've captured that (can be a separate `test_git_operations.py` file, as it happens on different level that the CLI commands)

Reply:
@PawelLipski stack trace:
```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Users/amalota/Projects/git-machete/tests/mockers.py:349: in launch_command
    cli.launch(list(args))
/Users/amalota/Projects/git-machete/git_machete/cli.py:616: in launch
    opt_no_detect_squash_merges=cli_opts.opt_no_detect_squash_merges))
/Users/amalota/Projects/git-machete/git_machete/client.py:1208: in fork_point
    branch, use_overrides, opt_no_detect_squash_merges=opt_no_detect_squash_merges)
/Users/amalota/Projects/git-machete/git_machete/client.py:1151: in __fork_point_and_containing_branch_pairs
    overridden_fp_hash = self.__get_overridden_fork_point(branch)
/Users/amalota/Projects/git-machete/git_machete/client.py:1622: in __get_overridden_fork_point
    override_data: ForkPointOverrideData = self.__get_fork_point_override_data(branch)
/Users/amalota/Projects/git-machete/git_machete/client.py:1588: in __get_fork_point_override_data
    if FullCommitHash.is_valid(self.__git.get_config_attr_or_none(to_key)):
/Users/amalota/Projects/git-machete/git_machete/git_operations.py:131: in is_valid
    and GitContext().is_commit_present_in_repository(revision)
/Users/amalota/Projects/git-machete/git_machete/git_operations.py:466: in is_commit_present_in_repository
    return True if self._run_git("rev-parse", f"--verify --quiet {revision}" + "^{commit}", allow_non_zero=True) == 0 else False
/Users/amalota/Projects/git-machete/git_machete/git_operations.py:211: in _run_git
    exit_code = utils.run_cmd("git", git_cmd, *args, **kwargs)
/Users/amalota/Projects/git-machete/tests/mockers.py:373: in mock_run_cmd
    [cmd] + list(args), stdout=subprocess.PIPE, stderr=subprocess.PIPE, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

input = None, timeout = None, check = False
popenargs = (['git', 'rev-parse', '--verify --quiet aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa^{commit}'],)
kwargs = {'allow_non_zero': True, 'stderr': -1, 'stdout': -1}

    def run(*popenargs, input=None, timeout=None, check=False, **kwargs):
        """Run command with arguments and return a CompletedProcess instance.
    
        The returned instance will have attributes args, returncode, stdout and
        stderr. By default, stdout and stderr are not captured, and those attributes
        will be None. Pass stdout=PIPE and/or stderr=PIPE in order to capture them.
    
        If check is True and the exit code was non-zero, it raises a
        CalledProcessError. The CalledProcessError object will have the return code
        in the returncode attribute, and output & stderr attributes if those streams
        were captured.
    
        If timeout is given, and the process takes too long, a TimeoutExpired
        exception will be raised.
    
        There is an optional argument "input", allowing you to
        pass a string to the subprocess's stdin.  If you use this argument
        you may not also use the Popen constructor's "stdin" argument, as
        it will be used internally.
    
        The other arguments are the same as for the Popen constructor.
    
        If universal_newlines=True is passed, the "input" argument must be a
        string and stdout/stderr in the returned object will be strings rather than
        bytes.
        """
        if input is not None:
            if 'stdin' in kwargs:
                raise ValueError('stdin and input arguments may not both be used.')
            kwargs['stdin'] = PIPE
    
>       with Popen(*popenargs, **kwargs) as process:
E       TypeError: __init__() got an unexpected keyword argument 'allow_non_zero'

/Users/amalota/.pyenv/versions/3.6.15/lib/python3.6/subprocess.py:423: TypeError